cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala

```diff
@@ -0,0 +1,127 @@
+  private def workflowSpecificEndpoint(workflowId: WorkflowId, endpoint: String) = s"$submitEndpoint/$workflowId/$endpoint"
+  def abortEndpoint(workflowId: WorkflowId) = workflowSpecificEndpoint(workflowId, "abort")
+  def statusEndpoint(workflowId: WorkflowId) = workflowSpecificEndpoint(workflowId, "status")
+  def metadataEndpoint(workflowId: WorkflowId) = workflowSpecificEndpoint(workflowId, "metadata")
+  lazy val backendsEndpoint = s"$submitEndpoint/backends"
+
+  import model.CromwellStatusJsonSupport._
+  import model.CromwellBackendsJsonSupport._
+
+  def submit(workflow: WorkflowSubmission)(implicit ec: ExecutionContext): Future[SubmittedWorkflow] = {
+    val sourceBodyParts = Map(
+      "wdlSource" -> Option(workflow.wdl),
+      "workflowInputs" -> workflow.inputs,
+      "workflowOptions" -> insertSecrets(workflow.options, workflow.refreshToken)
+    ) collect { case (name, Some(source)) => Multipart.FormData.BodyPart(name, HttpEntity(MediaTypes.`application/json`, ByteString(source))) }
+
+    val zipBodyParts = Map(
+      "wdlDependencies" -> workflow.zippedImports
+    ) collect { case (name, Some(file)) => Multipart.FormData.BodyPart.fromPath(name, MediaTypes.`application/zip`, file.path) }
+
+    val multipartFormData = Multipart.FormData((sourceBodyParts ++ zipBodyParts).toSeq : _*)
+    val requestEntity = multipartFormData.toEntity()
+
+    makeRequest[CromwellStatus](HttpRequest(HttpMethods.POST, submitEndpoint, immutable.Seq.empty[HttpHeader], requestEntity)) map { status =>
+      SubmittedWorkflow(WorkflowId.fromString(status.id), cromwellUrl, workflow)
+    }
+  }
+
+  def submitBatch(workflow: WorkflowBatchSubmission)(implicit ec: ExecutionContext): Future[Seq[SubmittedWorkflow]] = {
+    import DefaultJsonProtocol._
+
+    val sourceBodyParts = Map(
+      "wdlSource" -> Option(workflow.wdl),
+      "workflowInputs" -> Option(workflow.inputsBatch.mkString(start="[", sep=",", end="]")),
+      "workflowOptions" -> insertSecrets(workflow.options, workflow.refreshToken)
+    ) collect { case (name, Some(source)) => Multipart.FormData.BodyPart(name, HttpEntity(MediaTypes.`application/json`, ByteString(source))) }
+
+    val zipBodyParts = Map(
+      "wdlDependencies" -> workflow.zippedImports
+    ) collect { case (name, Some(file)) => Multipart.FormData.BodyPart.fromPath(name, MediaTypes.`application/zip`, file.path) }
+
+    val multipartFormData = Multipart.FormData((sourceBodyParts ++ zipBodyParts).toSeq : _*)
+    val requestEntity = multipartFormData.toEntity()
+
+    // Make a set of submissions that represent the batch (so we can zip with the results later):
+    val submissionSet = workflow.inputsBatch.map(inputs => WorkflowSubmission(workflow.wdl, Option(inputs), workflow.options, workflow.zippedImports, workflow.refreshToken))
+
+    makeRequest[List[CromwellStatus]](HttpRequest(HttpMethods.POST, batchSubmitEndpoint, immutable.Seq.empty[HttpHeader], requestEntity)) map { statuses =>
```

Comment:
Oh, uh, this is copy/paste from Centaur so I can't vouch for it either way... I'll git it a go the other way